Sources/Winget-AutoUpdate/functions/Invoke-LogRotation.ps1

```diff
@@ -1,55 +1,28 @@
-        .DESCRIPTION
-            Invoke-LogRotation handles log rotation
-        .NOTES
-            Author: Øyvind Kallstad (Minimized and changed for WAU 12.01.2023 by Göran Axel Johannesson)
-            URL: https://www.powershellgallery.com/packages/Communary.Logger/1.1
-            Date: 21.11.2014
-            Version: 1.0
-    #>
-
-    try {
-        # get current size of log file
-        $currentSize = (Get-Item $LogFile).Length
-
-        # get log name
-        $logFileName = Split-Path $LogFile -Leaf
-        $logFilePath = Split-Path $LogFile
-        $logFileNameWithoutExtension = [System.IO.Path]::GetFileNameWithoutExtension($logFileName)
-        $logFileNameExtension = [System.IO.Path]::GetExtension($logFileName)
-
-        # if MaxLogFiles is 1 just keep the original one and let it grow
-        if (-not($MaxLogFiles -eq 1)) {
+
+    # if MaxLogFiles is 1 just keep the original one and let it grow
+    if (-not($MaxLogFiles -eq 1)) {
+
+        try {
+            # get current size of log file
+            $currentSize = (Get-Item $LogFile).Length
+
+            # get log name
+            $logFileName = Split-Path $LogFile -Leaf
+            $logFilePath = Split-Path $LogFile
+            $logFileNameWithoutExtension = [System.IO.Path]::GetFileNameWithoutExtension($logFileName)
+            $logFileNameExtension = [System.IO.Path]::GetExtension($logFileName)
+
             if ($currentSize -ge $MaxLogSize) {
 
                 # construct name of archived log file
                 $newLogFileName = $logFileNameWithoutExtension + (Get-Date -Format 'yyyyMMddHHmmss').ToString() + $logFileNameExtension
+                Rename-Item -Path $LogFile -NewName $newLogFileName -Force -Confirm:$false
 
-                # copy old log file to new using the archived name constructed above
-                Copy-Item -Path $LogFile -Destination (Join-Path (Split-Path $LogFile) $newLogFileName)
-
-                # Create a new log file
-                try {
-                    Remove-Item -Path $LogFile -Force
-                    New-Item -ItemType File -Path $LogFile -Force
-                    #Set ACL for users on logfile
-                    $NewAcl = Get-Acl -Path $LogFile
-                    $identity = New-Object System.Security.Principal.SecurityIdentifier S-1-5-11
-                    $fileSystemRights = "Modify"
-                    $type = "Allow"
-                    $fileSystemAccessRuleArgumentList = $identity, $fileSystemRights, $type
-                    $fileSystemAccessRule = New-Object -TypeName System.Security.AccessControl.FileSystemAccessRule -ArgumentList $fileSystemAccessRuleArgumentList
-                    $NewAcl.SetAccessRule($fileSystemAccessRule)
-                    Set-Acl -Path $LogFile -AclObject $NewAcl
-                }
-                catch {
-                    Return $False
-                }
+                # create new file
+                Write-ToLog "New log file created"
```

Comment:
@KnifMelti just to be sure I didn't break anything? It seems better to me, but if you could confirm...